app/controllers/application_controller.rb

```diff
@@ -6,14 +6,14 @@ class ApplicationController < ActionController::Base
   # run these methods on page load
   before_filter :current_user
 
-  private
- 
-  # Finds the User with the ID stored in the session with the key
-  # :current_user_id This is a common way to handle user login in
-  # a Rails application; logging in sets the session value and
-  # logging out removes it.
+  helper_method :current_user
+
   def current_user
-    session[:id] ||= request.remote_ip.hash + rand(1000)
-    @_current_user = session[:id]
+    @current_user ||= unless @current_user 
```

Comment:
I dont like this rewrite. more code, removed comment, and it does the same thing